packages/admin/admin/src/EllipsisTooltip.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+
+import { Tooltip } from "./common/Tooltip";
+
+export interface EllipsisTooltipProps {
+    children?: React.ReactNode;
+}
+
+export const EllipsisTooltip = ({ children }: EllipsisTooltipProps) => {
+    const rootRef = React.useRef<HTMLDivElement>(null);
+    const contentRef = React.useRef<HTMLSpanElement>(null);
+
+    const [renderWithTooltip, setRenderWithTooltip] = React.useState(false);
+
+    const updateRenderWithTooltip = React.useCallback(() => {
+        if (rootRef.current && contentRef.current) {
+            setRenderWithTooltip(contentRef.current.offsetWidth > rootRef.current.clientWidth);
+        }
+        // The dependency array items must be `.current`, otherwise the callback will not be called, when the html-element is rendered.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [rootRef.current, contentRef.current]);
```

Comment:
You could define the function inside the useEffect for simplicity. Also, if you always render the tooltip, the refs wouldn't need to be in the dependency array